class.jetpack-xmlrpc-server.php

```diff
@@ -93,25 +127,37 @@ function verify_registration( $verify_secret ) {
 	function verify_action( $params ) {
 		$action = $params[0];
 		$verify_secret = $params[1];
+		$state = $params[2];
 
 		if ( empty( $verify_secret ) ) {
 			return $this->error( new Jetpack_Error( 'verify_secret_1_missing', sprintf( 'The required "%s" parameter is missing.', 'secret_1' ), 400 ) );
-		} else if ( !is_string( $verify_secret ) ) {
+		} else if ( ! is_string( $verify_secret ) ) {
 			return $this->error( new Jetpack_Error( 'verify_secret_1_malformed', sprintf( 'The required "%s" parameter is malformed.', 'secret_1' ), 400 ) );
 		}
 
+		if ( empty( $state ) ) {
+			return $this->error( new Jetpack_Error( 'state_missing', sprintf( 'The required "%s" parameter is missing.', 'state' ), 400 ) );
+		} else if ( ! ctype_digit( $state ) ) {
+			return $this->error( new Jetpack_Error( 'state_malformed', sprintf( 'The required "%s" parameter is malformed.', 'state' ), 400 ) );
```

Comment:
This is breaking the Jetpack debug. Unauthenticated requests for connected sites appear to return this error, which would lead to the site appearing to be problematic on jetpack.com/support/debug/ and the in-plugin debugger.